homeassistant/components/crownstone/light.py

```diff
@@ -0,0 +1,204 @@
+    for sphere in manager.cloud.cloud_data:
+        for crownstone in sphere.crownstones:
+            if crownstone.type in CROWNSTONE_INCLUDE_TYPES:
+                # Crownstone can communicate with Crownstone USB
+                if manager.uart and sphere.cloud_id == manager.usb_sphere_id:
+                    entities.append(CrownstoneEntity(crownstone, manager.uart))
+                # Crownstone can't communicate with Crownstone USB
+                else:
+                    entities.append(CrownstoneEntity(crownstone))
+
+    async_add_entities(entities)
+
+
+def crownstone_state_to_hass(value: int) -> int:
+    """Crownstone 0..100 to hass 0..255."""
+    return map_from_to(value, 0, 100, 0, 255)
+
+
+def hass_to_crownstone_state(value: int) -> int:
+    """Hass 0..255 to Crownstone 0..100."""
+    return map_from_to(value, 0, 255, 0, 100)
+
+
+class CrownstoneEntity(CrownstoneDevice, LightEntity):
+    """
+    Representation of a crownstone.
+
+    Light platform is used to support dimming.
+    """
+
+    _attr_should_poll = False
+    _attr_icon = "mdi:power-socket-de"
+
+    def __init__(self, crownstone_data: Crownstone, usb: CrownstoneUart = None) -> None:
+        """Initialize the crownstone."""
+        super().__init__(crownstone_data)
+        self.usb = usb
+        # Entity class attributes
+        self._attr_name = str(self.device.name)
+        self._attr_unique_id = f"{self.cloud_id}-{CROWNSTONE_SUFFIX}"
+
+    @property
+    def usb_available(self) -> bool:
+        """Return if this entity can use a usb dongle."""
+        return self.usb is not None and self.usb.is_ready()
+
+    @property
+    def brightness(self) -> int | None:
+        """Return the brightness if dimming enabled."""
+        return crownstone_state_to_hass(self.device.state)
+
+    @property
+    def is_on(self) -> bool:
+        """Return if the device is on."""
+        return crownstone_state_to_hass(self.device.state) > 0
+
+    @property
+    def supported_features(self) -> int:
+        """Return the supported features of this Crownstone."""
+        if self.device.abilities.get(DIMMING_ABILITY).is_enabled:
+            return SUPPORT_BRIGHTNESS
+        return 0
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """State attributes for Crownstone devices."""
+        attributes: dict[str, Any] = {}
+        # switch method
+        if self.usb_available:
+            attributes["switch_method"] = "Crownstone USB Dongle"
+        else:
+            attributes["switch_method"] = "Crownstone Cloud"
+
+        # crownstone abilities
+        attributes["dimming"] = ABILITY_STATE.get(
+            self.device.abilities.get(DIMMING_ABILITY).is_enabled
+        )
+        attributes["tap_to_toggle"] = ABILITY_STATE.get(
+            self.device.abilities.get(TAP_TO_TOGGLE_ABILITY).is_enabled
+        )
+        attributes["switchcraft"] = ABILITY_STATE.get(
```

Comment:
Are these attributes relevant without knowing the current main state of the light? If so, they should be separate (binary) sensors.

Reply:
`dimming` provides relevant information about the entity, though home assistant already has a way of recognizing this through the supported features. The other 2 attributes are less relevant and could be binary sensors.

Reply:
Ok. Yeah we don't need to double up the dimming information I think. 👍